### PR TITLE
Acceptance tests for vra7_deployment resource and fix for issue # 143

### PR DESCRIPTION
**Pull request by Prativa20**
1. Acceptance tests for vra7_deployment resource
2. Fix for https://github.com/vmware/terraform-provider-vra7/issues/143
During update and delete requests, fetch the correct request id from the
POST request's response header and wait until termination status to return
from the provider

Signed-off-by: Prativa Bawri<bawrip@vmware.com>

### CODE REVIEW

File: vrealize/resource.go [hunk 1]
```diff
@@ -128,8 +128,12 @@ func createResource(d *schema.ResourceData, meta interface{}) error {
 	//Set request ID
 	d.SetId(catalogRequest.ID)
 	//Set request status
-	d.Set(utils.RequestStatus, sdk.Submitted)
-	return waitForRequestCompletion(d, meta)
+	//d.Set(utils.RequestStatus, sdk.Submitted)
```
Comment (markpeek) [inline vrealize/resource.go]: Should this comment be removed?

File: vrealize/resource.go [hunk 2]
```diff
@@ -513,29 +497,26 @@ func waitForRequestCompletion(d *schema.ResourceData, meta interface{}) error {
...
-		requestStatus = d.Get(utils.RequestStatus).(string)
-		log.Info("Checking to see if resource is created. Status: %s.", requestStatus)
-		if requestStatus == sdk.Successful {
-			log.Info("Resource creation SUCCESSFUL.")
-			return nil
-		} else if requestStatus == sdk.Failed {
+		reqestStatusView, _ := vraClient.GetRequestStatus(requestID)
+		status := reqestStatusView.Phase
+		d.Set(utils.RequestStatus, status)
+		log.Info("Checking to see the status of the request. Status: %s.", requestStatus)
```
Comment (markpeek) [inline vrealize/resource.go:505]: At some point we should audit the log messages for content, level, and perhaps give a better indication what action it is checking. Don't need to fix right now but perhaps file an issue for future improvement.

**Review by markpeek — APPROVED** [review]
LGTM! But I believe you might be making a few other tweaks. When you're ready we should commit these changes and then follow up with any file name changes in a separate PR.

File: vrealize/resource.go (hunk 1 above)
Comment (Prativa20) [reply vrealize/resource.go]: yes, i agree!

File: vrealize/resource.go (hunk 2 above)
Comment (Prativa20) [reply vrealize/resource.go:505]: okay, I will open an issue for this one.

**Pull request merged**